bigchaindb/common/schema/transaction.yaml

```diff
@@ -154,7 +154,10 @@ definitions:
             additionalProperties: true
           uri:
             type: string
-            pattern: "^ni:///sha-256;([a-zA-Z0-9_-]{0,86})?(.+)$"
+            pattern: "^ni:///sha-256;([a-zA-Z0-9_-]{0,86})[?]\
```

Comment:
I am not sure I understand the repetition on the regex. Is it because you want to have a least one match or because you want to handle the case where you can have a `&`?

Have you considered using something like:
`"^ni:///sha-256;([a-zA-Z0-9_-]{0,86})[?](fpt=(ed25519|threshold)-sha-256(&)?|cost=[0-9]+(&)?|subtypes=ed25519-sha-256(&)?){1,3}$"`

This would eliminate the repetition